build/nuget-release.yml

```
@@ -0,0 +1,172 @@
+              targetFiles: 'appsettings.json'
+              encoding: 'auto'
+              writeBOM: true
+              actionOnMissing: 'fail'
+              keepToken: false
+              tokenPrefix: '#{'
+              tokenSuffix: '}#'
+          - task: DotNetCoreCLI@2
+            displayName: 'Run integration tests'
+            inputs:
+              command: test
+              projects: 'src/**/$(Project).Tests.Integration.csproj'
+              arguments: '--configuration $(Build.Configuration)'
+              nobuild: true
+              publishTestResults: true
+
+  - stage: Release
+    displayName: 'Release to MyGet'
+    dependsOn: IntegrationTests
+    condition: succeeded()
+    jobs:
+      - job: PushToNuGet
+        displayName: 'Push to NuGet'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Templates'
+              path: '$(Build.SourcesDirectory)/templates'
+          - task: GitHubRelease@0
+            displayName: 'Create GitHub Release (Conditional Step)'
+            inputs:
+              gitHubConnection: 'GitHub (arcus-automation - OAuth)'
+              repositoryName: 'arcus-azure/arcus.templates'
+              tagSource: manual
+              tag: 'v$(Build.BuildNumber)'
+              title: 'v$(Build.BuildNumber)'
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Template.WebApi/$(Build.BuildNumber))
+                ```shell
+                > Install-Package Arcus.Template.WebApi -Version $(Build.BuildNumber)
+                ```
+            condition: not(contains(variables['Build.BuildNumber'], '-'))
+          - task: GitHubRelease@0
+            displayName: 'Create GitHub Pre-Release (Conditional Step)'
+            inputs:
+              gitHubConnection: 'GitHub (arcus-automation - OAuth)'
+              repositoryName: 'azure-arcus/arcus.templates'
+              tagSource: manual
+              tag: 'v$(Build.BuildNumber)'
+              title: 'v$(Build.BuildNumber)'
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Template.WebApi/$(Build.BuildNumber))
+                ```shell
+                > Install-Package Arcus.Template.WebApi -Version $(Build.BuildNumber)
+                ```
+              isPreRelease: true
+              compareWith: 'lastRelease'
+            condition: contains(variables['Build.BuildNumber'], '-')
+          - task: NuGetCommand@2
+            displayName: 'Push to NuGet.org'
+            inputs:
+              command: push
+              packagesToPush: 'templates/*.nupkg'
+              nuGetFeedType: external
+              publishFeedCredentials: 'MyGet.org (Push)'
```

I've seen your comment but think we can already change this given it works fine so we can ship it!

```suggestion
              publishFeedCredentials: 'NuGet.org'
```